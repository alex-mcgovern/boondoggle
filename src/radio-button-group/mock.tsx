import type * as React from "react";

import type { RadioButton } from "../radio-button";

import { LOREM } from "../../mocks/LOREM.mock";

export const RADIO_BUTTONS_MOCK: Array<
	React.ComponentProps<typeof RadioButton>
> = [
	{
		description: LOREM.text_xxs,
		name: "radio_item_1",
		title: "Radio item 1",
		value: "radio_item_1",
	},
	{
		description: LOREM.text_xxs,
		name: "radio_item_2",
		title: "Radio item 2",
		value: "radio_item_2",
	},
	{
		description: LOREM.text_xxs,
		name: "radio_item_3",
		title: "Radio item 3",
		value: "radio_item_3",
	},
];
