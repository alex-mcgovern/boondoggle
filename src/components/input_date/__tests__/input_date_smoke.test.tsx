/**
 * @jest-environment jsdom
 */
import { render, waitFor } from "@testing-library/react";

import { InputDate } from "..";
import { LOREM } from "../../../../mocks/LOREM.mock";

import type { InputDateProps } from "..";

const PROPS: InputDateProps = {
    name: LOREM.name(),
    placeholder: LOREM.placeholder(),
};

const renderComponent = async ({ ...props }: InputDateProps) => {
    return waitFor(() => {
        return render(<InputDate {...props} />);
    });
};

describe("<InputDate />", () => {
    describe("Basic smoke tests", () => {
        test("should render without throwing", async () => {
            const { getByRole } = await renderComponent(PROPS);

            expect(getByRole("textbox")).not.toBeNull();
        });

        test("should render placeholder when present", async () => {
            const { getByRole } = await renderComponent({
                name: LOREM.name(),
                placeholder: LOREM.placeholder(),
            });

            expect((getByRole("textbox") as HTMLInputElement).placeholder).toBe(
                LOREM.placeholder()
            );
        });
    });
});
