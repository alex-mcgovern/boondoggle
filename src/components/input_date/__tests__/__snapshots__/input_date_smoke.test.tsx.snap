// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDate /> Basic smoke tests should match snapshot 1`] = `
<input
  class="slot_wrapper_inset.styles_offsetSlotLeft__1r5w8or3 input.styles_getInputStyles__3x32wp7 input.styles__3x32wp6 get_sprinkles_width_100%_mobile__eiij7bgl a11y_a11yDisabled__r8he5t1 input.styles_getInputStyles_size_md__3x32wp9 input.styles_size_md__3x32wp4 element_size_elementFontSize_md__1f6zrv4 element_size__1f6zrv1 get_sprinkles_fontSize_body_md__eiij7b28 get_sprinkles_lineHeight_body_md__eiij7b3w element_size_elementPadding_md__1f6zrva element_size__1f6zrv7 get_sprinkles_paddingLeft_spacing_1.5_mobile__eiij7bdo get_sprinkles_paddingRight_spacing_1.5_mobile__eiij7bel get_sprinkles_paddingTop_spacing_1_mobile__eiij7bff get_sprinkles_paddingBottom_spacing_1_mobile__eiij7bco input.styles__3x32wp1 get_sprinkles_height_spacing_5__eiij7b3j input_date.styles_inputDate__1dufx3p0"
  name="Name ipsum"
  placeholder="Placeholder ipsum"
  readonly=""
  value=""
/>
`;
